src/attr/validators.py

```diff
@@ -407,14 +407,16 @@ def deep_iterable(member_validator, iterable_validator=None):
     """
     A validator that performs deep validation of an iterable.
 
-    :param member_validator: Validator to apply to iterable members
+    :param member_validator: Validator(s) to apply to iterable members
     :param iterable_validator: Validator to apply to iterable itself
         (optional)
 
     .. versionadded:: 19.1.0
 
     :raises TypeError: if any sub-validators fail
     """
+    if isinstance(member_validator, list):
+        member_validator = _AndValidator(member_validator)
```

Comment:
I've just double-checked and if I see it correctly, the validator argument of fields uses `and_()` because that flattens other lists/ands.

Reply:
Didn't quite understand this, could you elaborate ?

Reply:
My point is that you're directly instantiating the `_AndValidator` while the equivalent logic otherwhere uses `and_` because that has extra logic:

https://github.com/python-attrs/attrs/blob/b75436630fafde95ba2954052c5c4d7d4f506b28/src/attr/_make.py#L295-L296

Which reminds me that you should also check for tuple too. :)

Reply:
ahh I see. I was trying to mimic how this same logic was implemented for the `optional` validator. I'm happy to change it to `and_` if thats better but then we should probably add in a issue to update the `optional` validator too (I can make the issue if you'd like)
https://github.com/python-attrs/attrs/blob/main/src/attr/validators.py#L278-L293

Reply:
Haven't updated this yet, will hold off till we have discussed^

Reply:
Hm yes, optional should be fixed too, but _not_ as part of this PR. But make this one as in _make please.

Reply:
Sounds good, I'll open a separate Issue for that.

Updated this PR with the `and_` validator